gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+        return sc, conf, rpc_mgr
+
+    def _get_GenericConfigRpcManager_object(self, conf, sc):
+        agent = gc.GenericConfigRpcManager(sc, conf)
+        return agent, sc
+
+    @mock.patch(__name__ + '.FakeObjects.nqueue')
+    @mock.patch(__name__ + '.FakeObjects.drivers')
+    def _get_GenericConfigEventHandler_object(self, sc, rpcmgr, drivers, nqueue):
+        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
+        return agent
+
+    def _get_FWaasRpcManager_object(self, conf, sc):
+        agent = fw.FWaasRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_generic_event_creation(self, operation, method, bulk=False):
+        sc, conf, rpc_mgr = self._get_ConfiguratorRpcManager_object()
+        agent, sc = self._get_GenericConfigRpcManager_object(conf, sc)
+        
+        if bulk:
+            request_data = self.fo.fake_request_data_generic_bulk()
+            request_data1 = self.fo.fake_request_data_generic_bulk()
+        else:
+            request_data = self.fo.fake_request_data_generic_single()
+            request_data1 = self.fo.fake_request_data_generic_single()
+
+        if (method == 'CONFIGURE_ROUTES' or method == 'CLEAR_ROUTES'):
+            request_data1['config'][0]['resource'] = 'routes'
+            request_data['config'][0]['resource'] = 'routes'
+
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event, \
+            mock.patch.object(rpc_mgr,
+                              '_get_service_agent_instance',
+                              return_value=agent):
+            if operation == 'create':
+                rpc_mgr.create_network_device_config(
+                                    self.fo.context, request_data)
+            elif operation == 'delete':
+                rpc_mgr.delete_network_device_config(
+                                    self.fo.context, request_data)
+
+            context = request_data1['config'][0]['kwargs']['context']
+            context.update({'notification_data': {}})
+            context.update({'resource': request_data1['config'][0]['resource']})
+            del request_data1['config'][0]['kwargs']['context']
+            kwargs = request_data1['config'][0]['kwargs']
+            if bulk:
+                sa_info_list = self.fo.fake_sa_info_list()
+                if operation == 'delete':
+                    sa_info_list[0]['method'] = 'clear_interfaces'
+                    sa_info_list[1]['method'] = 'clear_routes'
+                args_dict = {
+                         'sa_info_list': sa_info_list,
+                         'notification_data': {}
+                        }
+            else:
+                args_dict = {'context': context,
+                             'kwargs': kwargs}
+            mock_sc_event.assert_called_with(id=method, data=args_dict)
```

Comment:
This won't invoke RPC methods

Regards,
Deepak

On Fri, Mar 11, 2016 at 9:38 AM, njagadish <notifications@github.com> wrote:

> In gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py
> <https://github.com/oneconvergence/group-based-policy/pull/80#discussion_r55787642>
> :
>
> > +            context.update({'resource': request_data1['config'][0]['resource']})
> > +            del request_data1['config'][0]['kwargs']['context']
> > +            kwargs = request_data1['config'][0]['kwargs']
> > +            if bulk:
> > +                sa_info_list = self.fo.fake_sa_info_list()
> > +                if operation == 'delete':
> > +                    sa_info_list[0]['method'] = 'clear_interfaces'
> > +                    sa_info_list[1]['method'] = 'clear_routes'
> > +                args_dict = {
> > +                         'sa_info_list': sa_info_list,
> > +                         'notification_data': {}
> > +                        }
> > +            else:
> > +                args_dict = {'context': context,
> > +                             'kwargs': kwargs}
> > +            mock_sc_event.assert_called_with(id=method, data=args_dict)
>
> We mocked only new_event and post_event. We also have agent RPC method
> invocation. Shouldn't that be mocked too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oneconvergence/group-based-policy/pull/80/files#r55787642>
> .
>